### PR TITLE
Android gradle plugin 7.0.2 compatible + Support for Android 12 target build

### PR DESCRIPTION
**Pull request by michaldabrowskistorm**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,12 +4,14 @@
...
     <application
         android:allowBackup="true"
+        android:fullBackupOnly="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme" >
         <activity android:name=".MainActivity"
+            android:exported="true"
```
Comment (fnabrdal) [inline app/src/main/AndroidManifest.xml:14]: why exported?

File: build.gradle [hunk 2]
```diff
@@ -19,9 +19,17 @@ allprojects {
         google()
         jcenter()
 
-        maven {
-            url 'https://www.myget.org/F/storyteller-android-sdk/maven/'
+        exclusiveContent{
```
[the remark this answers is not in the document]
Comment (michaldabrowskistorm) [reply build.gradle:22]: The reasoning behind that
https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:repository-content-filtering

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (michaldabrowskistorm) [reply app/src/main/AndroidManifest.xml:14]: Changes in 12
https://developer.android.com/about/versions/12/behavior-changes-12#security

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (DaveMiscampbell) [reply build.gradle:22]: How long has this been a thing in Gradle for? How common is it? I've never seen another open-source library recommend doing it this way for example. My suggestion would potentially be to include the "easy" way and then add a comment with this method displayed and a link to the docs - shows we care about the security aspect but doesn't confuse anybody with a more basic understanding

[the remark this answers is not in the document]
Comment (michaldabrowskistorm) [reply build.gradle:22]: Feature of content filtering itself is pretty old https://docs.gradle.org/5.6.4/userguide/declaring_repositories.html#declaring_a_repository_filter

The exclusive content was added 6.x

I would agree that is not common knowledge (I've learned that when i was forced to cut cost of CI/CD execution for sdk i was developing),  but I can significantly decrease build times - especially if you have other's party SDK sluggish maven on your repository list.

As mentioned I will revert that section to the basic usage and I will add section in the docs about alternative for more complex build setup (for sure abc could use this approach 😄  )

**Review by fnabrdal — APPROVED** [review]
Good work! We can merge it to develop.

**Pull request merged**